### PR TITLE
Added Matrix-Chain-Multiplication in Python.

### PR DESCRIPTION
**Pull request by Pranav016**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Dynamic Programming problem of Matrix-Chain-Multiplication in Python.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Solves #1474

### CODE REVIEW

File: Python/dp/Matrix-Chain-Multiplication.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import sys
```
Comment (HarshCasper) [inline Python/dp/Matrix-Chain-Multiplication.py:13]: Add a beginner pitch enclosed in a multiline string detailing about the Code/Algorithm

File: Python/dp/Matrix-Chain-Multiplication.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def mcm(arr, i, j, dp):
+    '''
```
Comment (HarshCasper) [inline Python/dp/Matrix-Chain-Multiplication.py:17]: Better use Docstrings for the purpose. 

```
"""
This is an example of Google style.

Args:
    param1: This is the first param.
    param2: This is a second param.

Returns:
    This is a description of what is returned.

Raises:
    KeyError: Raises an exception.
"""
```

File: Python/dp/Matrix-Chain-Multiplication.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    Summary line:
+    This function helps us to find the minimum cost
+    required to multiply matrices.
+    '''
+    if i >= j:
+        return 0
+    ans = sys.maxsize
+    for k in range(i, j):
```
Comment (HarshCasper) [inline Python/dp/Matrix-Chain-Multiplication.py:35]: Add proper Comments explaining the Code

File: Python/dp/Matrix-Chain-Multiplication.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        ans = min(ans, cost)
+    return ans
+
+if __name__ == "__main__":
+    n = int(input())
+    arr = [int(i) for i in input().split()]
+    dp = [[-1 for j in range(n+1)]for i in range(n+1)]
+    print(mcm(arr, 1, n, dp))
+
+# Sample input-
```
Comment (HarshCasper) [inline Python/dp/Matrix-Chain-Multiplication.py]: Enclose this in a Multi-Line String

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Code looks good to me 👍

File: Python/dp/Matrix-Chain-Multiplication.py (hunk 3 above)
Comment (Pranav016) [reply Python/dp/Matrix-Chain-Multiplication.py:35]: Sure, I'll add comments.

File: Python/dp/Matrix-Chain-Multiplication.py (hunk 2 above)
Comment (Pranav016) [reply Python/dp/Matrix-Chain-Multiplication.py:17]: ok 👍

**Comment by Pranav016** [comment]
Can you please review this? @iamrajiv @atarax665 @shraddhavp
Thanks

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**